### PR TITLE
Feat: add support of READONLY and READWRITE

### LINKED ISSUE #2166: Add support of the command READONLY & READWRITE
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/kvrocks/issues) and found no similar issues.

### Motivation

1. redis cluster utilize [readonly](https://redis.io/commands/readonly/) to support reading from slave node
2. go-redis will send a 'readonly' command to server when 'ReadOnly' option is true during initialisation which would incur an error response from kvrocks 'ERR unknown command readonly' because kvrocks doesn't support command 'readonly'
3. this is important because without reading from slave a shard  is not readable during failover which is not acceptable for some businesses

### Solution

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by LiuYuHui**
closes #2166 
local test pass
image
image

### CODE REVIEW

File: src/cluster/cluster.h [hunk 1]
```diff
@@ -111,4 +118,6 @@ class Cluster {
 
   std::map<int, std::string> migrated_slots_;
   std::set<int> imported_slots_;
+
+  ClusterMode cluster_mode_{ClusterMode::READONLY};
```
Comment (caipengbo) [inline src/cluster/cluster.h]: These two commands should be in the **connection** dimension.

File: src/commands/cmd_cluster.cc [hunk 2]
```diff
@@ -292,8 +292,25 @@ static uint64_t GenerateClusterFlag(const std::vector<std::string> &args) {
   return 0;
 }
 
+class CommandReadOnly : public Commander {
+ public:
+  Status Execute(Server *srv, Connection *conn, std::string *output) override {
+    *output = redis::SimpleString("READONLY");
```
Comment (caipengbo) [inline src/commands/cmd_cluster.cc]: The reply should be `OK`.

File: src/commands/cmd_cluster.cc [hunk 3]
```diff
@@ -292,8 +292,25 @@ static uint64_t GenerateClusterFlag(const std::vector<std::string> &args) {
...
+    return srv->cluster->SetClusterMode(Cluster::ClusterMode::READONLY);
+  }
+};
+
+class CommandReadWrite : public Commander {
+ public:
+  Status Execute(Server *srv, Connection *conn, std::string *output) override {
+    *output = redis::SimpleString("READWRITE");
```
Comment (caipengbo) [inline src/commands/cmd_cluster.cc]: The reply should be `OK`.

File: src/cluster/cluster.h (hunk 1 above)
Comment (LiuYuHui) [reply src/cluster/cluster.h]: thanks, fixed it.

File: src/commands/cmd_cluster.cc (hunk 2 above)
Comment (LiuYuHui) [reply src/commands/cmd_cluster.cc]: thanks, fixed it.

File: src/server/redis_connection.h [hunk 4]
```diff
@@ -45,6 +45,7 @@ class Connection : public EvbufCallbackBase<Connection> {
     kCloseAfterReply = 1 << 6,
     kCloseAsync = 1 << 7,
     kMultiExec = 1 << 8,
+    KReadWrite = 1 << 9,
```
Comment (jihuayu) [inline src/server/redis_connection.h]: The default state of the flag be disabled? Should it be `KReadOnly` here?

**Review by jihuayu — COMMENTED** [review]
Thanks! Good job!

File: src/server/redis_connection.h (hunk 4 above)
Comment (LiuYuHui) [reply src/server/redis_connection.h]: It seems that the original logic is `KReadOnly` ==1, and the default flag is 0, so I choose `KReadWrite` here, which means the default option is `KReadWrite` ==0 (`KReadOnly` ==1).

Comment (jihuayu) [reply src/server/redis_connection.h]: What is the logic for Redis? The original logic may be wrong.
We need to align with Redis, not with the original.

Comment (LiuYuHui) [reply src/server/redis_connection.h]: https://redis.io/commands/readonly/
> Normally replica nodes will redirect clients to the authoritative master for the hash slot involved in a given command, however clients can use replicas in order to scale reads using the READONLY command.

I see, Redis default option is `READWRTIE`, will change it .

**Review by jihuayu — APPROVED** [review]
Thank you!

**Review by caipengbo — APPROVED** [review]

**Review by git-hulk — APPROVED** [review]

**Comment by git-hulk** [comment]
@LiuYuHui Thanks for your contribution.

**Pull request merged**